client-js/app/scripts/registry.ts

```diff
@@ -9,182 +11,10 @@ export interface RegistryListing {
   vals: Array<string>;
 }
 
-export interface RegistryApi {
```

Comment:
It's the same distinction as interface and class in java; an interface is just a collection of methods, but you can't actually create instances of an interface. A class is something you can create instances of, and it actually implements those methods. Previously, we had `RegistryService` and `RegistryServiceJsonRpc`, which both implemented the `RegistryApi` interface, so if some piece of code required a `RegistryApi` to work, you could pass an instance of either of those classes and it would be perfectly happy because the both satisfy the interface that it depends on. In a case like that where there are multiple implementations, it's clear that you want to separate the interface from the implementation, but even if we only have one implementation (the json rpc version), it's nice to have this separation. For example, if we want to test some code that needs a `RegistryApi`, we can write a little test class that implements this interface but doesn't actually make remote calls to a server and the code under test will be none the wiser.